src/file-queue.ts

```diff
@@ -27,31 +27,46 @@ export function makeFileQueue() {
   }
 }
 
-function indexOfFileInQueue(fileQueue: FileQueue, file: File) {
-  return binarySearch(fileQueue, fileInQueue => {
-    return file === fileInQueue
-      ? 0
-      : file.lastTouchedTimestamp - fileInQueue.lastTouchedTimestamp ||
+/**
+ * Uses binary search to find the index of a given file in a given fileQueue.
+ *
+ * @param fileQueue Array of File objects sorted by lastTouchTimestamp
+ * @param file The File object for which to search.
+ */
+function indexOfFileInQueue(fileQueue: File[], file: File) {
+  let l = 0
+  let r = fileQueue.length - 1
+
+  while (l <= r) {
+    // Use bit shift over division to avoid floats
+    let index = (l + r) >> 1
+
+    // File in queue to compare
+    const fileInQueue = fileQueue[index]
+
+    // Direction to continue binary search. Zero is a match.
+    // -1 is to search lower and 1 is to search higher.
+    const direction =
+      file === fileInQueue
+        ? 0
+        : file.lastTouchedTimestamp - fileInQueue.lastTouchedTimestamp ||
           // Fallback to lexigraphical comparison (if timestamps match)
+          // TODO: This solution requires that lexigraphical comparison is
+          // used on insert (queue method).
           (file.filename > fileInQueue.filename ? 1 : -1)
```

Comment:
I'd like to point out this line as it may not give correct results. What is needed to fix this is to add lexicographical sorting to the queue when timestamps match (rare condition). However, I left this work incomplete to open discussion to an alternative solution to matching timestamps.

Reply:
Added lexicographical sorting to the queue function with this commit: 1a7575908def9174b6d0eccd5a7ac7fe4ddbc0c8